.github/workflows/build.yml

```diff
@@ -0,0 +1,349 @@
+
+    - name: Extract freetype
+      run: |
+        tar xf ${{ steps.freetype.outputs.filepath }}
+        move freetype-2.13.3 freetype
+
+    - name: Download lua
+      id: lua
+      uses: mercury233/action-cache-download-file@v1.0.0
+      with:
+        url: https://www.lua.org/ftp/lua-5.4.7.tar.gz
+
+    - name: Extract lua
+      run: |
+        tar xf ${{ steps.lua.outputs.filepath }}
+        move lua-5.4.7 lua
+
+    - name: Download sqlite
+      id: sqlite
+      uses: mercury233/action-cache-download-file@v1.0.0
+      with:
+        url: https://www.sqlite.org/2025/sqlite-amalgamation-3490100.zip
+
+    - name: Extract sqlite
+      run: |
+        7z x ${{ steps.sqlite.outputs.filepath }}
+        move sqlite-amalgamation-3490100 sqlite3
+
+    - name: Download irrKlang
+      id: irrKlang
+      uses: mercury233/action-cache-download-file@v1.0.0
+      with:
+        url: https://www.ambiera.at/downloads/irrKlang-32bit-1.6.0.zip
+
+    - name: Extract irrKlang
+      run: |
+        7z x ${{ steps.irrKlang.outputs.filepath }}
+        move irrKlang-1.6.0 irrKlang
+
+    - name: Download irrlicht
+      run: |
+        git clone --depth=1 https://github.com/mercury233/irrlicht
+
+    - name: Check DirectX SDK
+      id: dxsdk
+      uses: actions/cache@v4
+      with:
+        key: dxsdk
+        path: DXSDK
+
+    - name: Download DirectX SDK
+      if: steps.dxsdk.outputs.cache-hit != 'true'
+      id: dxsdk-download
+      uses: mercury233/action-cache-download-file@v1.0.0
+      with:
+        url: https://download.microsoft.com/download/a/e/7/ae743f1f-632b-4809-87a9-aa1bb3458e31/DXSDK_Jun10.exe
+
+    - name: Install DirectX SDK
+      if: steps.dxsdk.outputs.cache-hit != 'true'
+      run: |
+        7z x ${{ steps.dxsdk-download.outputs.filepath }} -aoa
+
+    - name: Set DirectX SDK environment variable
+      run: |
+        $dxsdkPath = Resolve-Path 'DXSDK'
+        "DXSDK_DIR=$($dxsdkPath.ProviderPath)\" | Out-File -FilePath $env:GITHUB_ENV -Append
+
+    - name: Copy premake files
+      run: |
+        xcopy /E premake\* .
+
+    - name: Use premake to generate Visual Studio solution (2019)
+      if: matrix.os == 'windows-2019'
+      run: |
+        .\premake5.exe vs2019 --winxp-support
+
+    - name: Use premake to generate Visual Studio solution (2022)
+      if: matrix.os == 'windows-2022'
+      run: |
+        .\premake5.exe vs2022
+
+    - name: Add msbuild to PATH
+      uses: microsoft/setup-msbuild@v2
+
+    - name: Build solution
+      run: |
+        MSBuild.exe build\YGOPro.sln /m /p:Configuration=Release
+
+    - name: Upload build artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: YGOPro-${{ matrix.os }}
+        path: |
+          bin/release/YGOPro.exe
+
+  build-linux:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+#          - ubuntu-20.04
+          - ubuntu-22.04
+          - ubuntu-24.04
```

Comment:
How about Debian? In addition, I don't think we should drop Ubuntu 20.04 because it's still in support period.

Reply:
GitHub [don't have](https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories) Debian runner, and Ubuntu 20 runner will be removed in [the next month](https://github.com/actions/runner-images/issues/11101).